cloud-engineering-with-gcp/python/content/lab-02/README.md

```diff
@@ -0,0 +1,199 @@
+# Deploy a Static Website
+
+In the second lab of this workshop, we're going to deploy some static HTML files to a GCP Storage bucket.
+
+## Step 1 &mdash; Create a Bucket
+
+We'll first create the GCP storage bucket that will store our HTML files. Before we do that, we need to import the GCP provider into our project
+
+```python
+import pulumi_gcp as gcp
+```
+
+Now that we've imported our GCP provider, we can create our bucket:
+
+```python
+bucket = gcp.storage.Bucket(
+    "website",
+    location="US"
+)
+```
+
+You'll notice we've set a location parameter for our bucket.
+
+At this stage, your `__main.py__` file should match the following:
+
+```python
+"""A Python Pulumi program"""
+
+import pulumi
+import pulumi_gcp as gcp
+
+bucket = gcp.storage.Bucket(
+    "website",
+    location="US"
+)
+```
+
+## Step 2 &mdash; Configure the ACLs for the Bucket Object
+
+When we upload the HTML files to our bucket, we want them to be publicly accessible. In order to make sure every file we place in the bucket gets desired accessibility, we need to set a default ACL.
+
+Create a default ACL:
+
+```python
+acl = gcp.storage.DefaultObjectAccessControl(
+    'website',
+    bucket=bucket.name,
+    role="READER",
+    entity="allUsers"
+)
+```
+
+At this stage, your `__main.py__` file should match the following:
+
+```python
+"""A Python Pulumi program"""
+
+import pulumi_gcp as gcp
+
+bucket = gcp.storage.Bucket(
+    "website",
+    location="US"
+)
+
+acl = gcp.storage.DefaultObjectAccessControl(
+    'website',
+    bucket=bucket.name,
+    role="READER",
+    entity="allUsers"
+)
+```
+
+## Step 3 &mdash; Upload your Objects
+
+Now we need to upload the files that comprise our website so we can view them. Because Pulumi uses real programming languages, we can use constructs like `for` loops. Let's use a `for` loop to iterate over the files in the `wwwroot` directory in this repo and upload them using the `BucketObject` resource.
+
+First, we need to import the `os` Python library. Note that a major advantage of Pulumi's design of using real programming languages is that we can make use of both standard libraries and external packages when defining our infrastructure.
+
+Add the following statement near the top of your `__main__.py` near your other imports:
+
+```python
+import os
+```
+
+```python
+for file in ["404.html", "index.html"]:
```

Comment:
Why are you using a hardcoded list here instead of either a path search or a glob search?

Reply:
Will be fixed in the next commit.